Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM python:3.10-slim AS builder
+
+WORKDIR /app
+
+RUN pip install poetry==1.2.1 && poetry config virtualenvs.in-project true
+
+ADD pyproject.toml poetry.lock /app/
+RUN poetry install --no-root --without=dev --no-ansi
+RUN poetry install --only-root --no-ansi
```

Comment:
Now that I'm lookin at it, I'm not even sure I need to install the root project.

I thought that separating will cache dependencies so that when I change the code the dependencies stage will not run again.